DEER

mstojens opened the pull request:
Revisited the original write up with a week of objectivity and tied up the remaining loose ends.

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -49,13 +49,32 @@ author:
 
-This document...
+This document defines Discovery of Equivalent Encrypted Resolvers (DEER), a mechanism for DNS
+clients to query a resolver over unencrypted DNS for encrypted DNS configuration. It is
```
```suggestion
clients to use unencrypted DNS to discover a resolver's encrypted DNS configuration. It is
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -49,13 +49,32 @@ author:
-This document...
+This document defines Discovery of Equivalent Encrypted Resolvers (DEER), a mechanism for DNS
+clients to query a resolver over unencrypted DNS for encrypted DNS configuration. It is
+designed to be agnostic of different forms of DNS encryption for future flexibility. It
```
```suggestion
designed to be agnostic to different forms of DNS encryption for future flexibility. It
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -49,13 +49,32 @@ author:
+This document defines Discovery of Equivalent Encrypted Resolvers (DEER), a mechanism for DNS
+clients to query a resolver over unencrypted DNS for encrypted DNS configuration. It is
+designed to be agnostic of different forms of DNS encryption for future flexibility. It
+is also designed to provide assurance that the encrypted server connection is controlled
```
```suggestion
is also designed to ensure that the encrypted server connection is controlled
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -49,13 +49,32 @@ author:
-Stuff
+When DNS clients wish to use encrypted protocols such as DNS-over-TLS (DoT) {{!RFC7858}}
+or DNS-over-HTTPS (DoH) {{!RFC8484}}, they require additional information beyond the IP
+address of the DNS server. However, it is common for DNS clients to only learn a
```
```suggestion
address of the DNS server, such as the resolver's hostname. However, it is common for DNS clients to only learn a
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -49,13 +49,32 @@ author:
+When DNS clients wish to use encrypted protocols such as DNS-over-TLS (DoT) {{!RFC7858}}
+or DNS-over-HTTPS (DoH) {{!RFC8484}}, they require additional information beyond the IP
+address of the DNS server. However, it is common for DNS clients to only learn a
+resolver's IP address during configuration. Such mechanisms include network advertisement
```
```suggestion
resolver's IP address during configuration. Such mechanisms include network provisioning
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -49,13 +49,32 @@ author:
+or DNS-over-HTTPS (DoH) {{!RFC8484}}, they require additional information beyond the IP
+address of the DNS server. However, it is common for DNS clients to only learn a
+resolver's IP address during configuration. Such mechanisms include network advertisement
+such as DHCP and RA as well as manual configuration.
```
```suggestion
protocols like DHCP and IPv6 Router Advertisements, as well as manual configuration.
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -49,13 +49,32 @@ author:
+such as DHCP and RA as well as manual configuration.
+
+This document addresses encrypted DNS resolver discovery with two goals in mind:
+enable discovery from only an IP address, and allow clients to confirm the encrypted
```
```suggestion
enable discovery when only an IP address is known, and allow clients to confirm that the encrypted
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -49,13 +49,32 @@ author:
+
+This document addresses encrypted DNS resolver discovery with two goals in mind:
+enable discovery from only an IP address, and allow clients to confirm the encrypted
+server connection they establish is the same entity they were querying without
```
```suggestion
resolver they connect to is the same entity as the known resolver that did not use
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -49,13 +49,32 @@ author:
+
+For DNS servers that do not support encryption, their encrypted connection configuration
+can be requested by a new special use domain name (SUDN). For DNS servers that do support
+encryption, this configuration can be requested by their own domain name.
```
```suggestion
encryption, this configuration can be requested based on a query for their own domain name.
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -65,8 +84,256 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+discovered resolver and SHOULD suppress queries for Equivalent Encrypted Resolvers against
+this resolver for the TTL of the negative or invalid response and continue to use the original resolver.
+
+### Why both IP addresses are required
```
```suggestion
### Rationale for Validating Both Resolver Addresses 
```

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -65,8 +84,256 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+Future versions or other documents may define additional keys for discovery of resolvers
+using other encrypted transports.
+
+#### SVCB Records for Equivalent Encrypted Resolvers using DoH
```
We may not want to nest so deeply, but can clean up later

mstojens replied:
Agreed. I have no attachment to the current structure versus a flatter one. I just chose something to prevent writer's deadlock. The same is true of in-section security comments versus a security consideration section.

tfpauly commented on draft-pauly-add-deer.md:
```diff
@@ -65,8 +84,256 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+This enables a resolver to modify its answer to match the name being queried. It also
+enables resolvers to recursively answer the query if it is for a name it is not authoritative
+for. An example where this would be useful is when a client has DoH and DoT configuration
+for resolver Foo but only DoH configuration for resolver Bar. If DoH is being blocked on
```
Maybe use an `example.com` thing instead of Foo?

mstojens replied:
Good call.

tfpauly reviewed (approved):
Thanks! I've left a few nits. Please integrate as you see fit, and then merge. After that, I'll do a PR on top of this that updates some of the SVCB bits to adopt Ben Schwartz's document.

mstojens commented:
> Thanks! I've left a few nits. Please integrate as you see fit, and then merge. After that, I'll do a PR on top of this that updates some of the SVCB bits to adopt Ben Schwartz's document.

Thank you! I've addressed the nits; they all made sense. Will merge now.
